cli/pkg/validate/entity/entity_validator.go

```diff
@@ -0,0 +1,67 @@
+package entity
```

Comment:
One concern with this framework is that there is no clear separation of which part if generic and which part is specific to the Data Catalog. If we're looking to extend the CLI to other resources, they would probably need a similar framework for validations, with many common functionalities (e.g checking for unique ids per type, validate references etc).

This approach looks like it's operating on the DataCatalog struct. I get the feeling this dependency is forced by the loader which might need a similar generic framework for parsing YAMLs into structs while considering a common structure like entity groups with their metadata, entity ids, using references etc.